source/plugins/go/src/oms.go

```diff
@@ -1341,31 +1342,35 @@ func PostDataHelper(tailPluginRecords []map[interface{}]interface{}) int {
 			msgpBytes = msgp.AppendInt64(msgpBytes, batchTime)
 			msgpBytes = msgp.AppendMapStrStr(msgpBytes, fluentForward.Entries[entry].Record)
 		}
-		if IsWindows && IsGenevaLogsIntegrationEnabled {
-			if ContainerLogNamedPipe == nil {
-				Log("Error::AMA:: The connection to named pipe was nil. re-connecting...")
-				CreateWindowsNamedPipesClient(getGenevaWindowsNamedPipeName())
-			}
-			if ContainerLogNamedPipe == nil {
-				Log("Error::AMA::Cannot create the named pipe connection")
-				ContainerLogTelemetryMutex.Lock()
-				defer ContainerLogTelemetryMutex.Unlock()
-				ContainerLogsWindowsAMAClientCreateErrors += 1
-				return output.FLB_RETRY
+
+		if IsWindows {
+			var datatype string
+			if ContainerLogSchemaV2 {
+				datatype = ContainerLogV2DataType
+			} else {
+				datatype = ContainerLogDataType
 			}
-			Log("Info::AMA::Starting to write container logs to named pipe")
-			deadline := 10 * time.Second
-			ContainerLogNamedPipe.SetWriteDeadline(time.Now().Add(deadline))
-			n, err := ContainerLogNamedPipe.Write(msgpBytes)
-			if err != nil {
-				Log("Error::AMA::Failed to write to AMA %d records. Will retry ... error : %s", len(msgPackEntries), err.Error())
-				ContainerLogTelemetryMutex.Lock()
-				defer ContainerLogTelemetryMutex.Unlock()
-				ContainerLogsSendErrorsToWindowsAMAFromFluent += 1
-				return output.FLB_RETRY
+			if CreateGenevaOr3PNamedPipe(&ContainerLogNamedPipe, datatype, &ContainerLogsWindowsAMAClientCreateErrors, IsGenevaLogsIntegrationEnabled, &MdsdContainerLogTagRefreshTracker) {
```

Comment:
why do we need create Namedpipe for every record?  We should only create if its not created already, right?

Reply:
Yes will change the name of the function but we dont create this for every record.